aep/general/0151/aep.md.j2

```diff
@@ -7,41 +7,89 @@ return some kind of promise to the user, and allow the user to check back in
-`202 Accepted` along with an `Operation`, or _never_ do so. A service **must
-not** return a `200 OK` response with the result if it is "fast enough", and
-`202 Accepted` if it is not fast enough, because such behavior adds significant
-burdens for clients.
+A single method defined in an API surface **must** either _always_ return an
+`Operation`, or _never_ do so. As an example, a long-running operation **must
+not** return a response with the result of the operation if the operation
+completes quickly.
 
-**Note:** User expectations can vary on what is considered "a significant
-amount of time" depending on what work is being done. A good rule of thumb is
-10 seconds.
+### Interface Definitions
 
-### Operation representation
+{% tab proto %}
 
-The response to a long-running request **must** be an [`Operation`][Operation].
+When using protocol buffers, the common component
+[`aep.api.Operation`][aep.api.Operation] is used.
 
-{% tab proto %}
+{% sample 'lro.proto', 'rpc WriteBook' %}
 
-Protocol buffer APIs **must** use the common component
-[`aep.api.Operation`][aep.api.Operation].
+- The response type **must** be `aep.api.Operation`. The `Operation` proto
+  definition **should not** be copied into individual APIs; prefer to use a
+  single copy (in monorepo code bases), or remote dependencies via a tool like
+  [Buf][buf.build].
+  - The response **must not** be a streaming response.
+- The method **must** include a [`aep.api.operation_info`][lro] annotation,
+  which **must** define both response and metadata types.
+  - The response and metadata types **must** be defined in the file where the
+    RPC appears, or a file imported by that file.
+  - If the response and metadata types are defined in another package, the
+    fully-qualified message name **must** be used.
+  - The response type **should not** be `google.protobuf.Empty` (except for
+    [`Delete`][aep-135] methods), unless it is certain that response data will
+    _never_ be needed. If response data might be added in the future, define an
+    empty message for the RPC response and use that.
+  - The metadata type is used to provide information such as progress, partial
+    failures, and similar information on each `GetOperation` call. The metadata
+    type **should not** be `google.protobuf.Empty`, unless it is certain that
+    metadata will _never_ be needed. If metadata might be added in the future,
+    define an empty message for the RPC metadata and use that.
+- APIs with messages that return `Operation` **must** implement the
+  `GetOperation` method of the [`Operations`][lro] service, and **may**
+  implement the other methods defined in that service. Individual APIs **must
+  not** define their own interfaces for long-running operations to avoid
+  inconsistency.
 
 {% tab oas %}
 
-OpenAPI services **must** use this [`JSON Schema Operation`][JSON Schema
-Operation] schema.
+{% sample 'lro.oas.yaml', 'paths' %}
```

Comment:
**thought:** Should the Operation part of the lro.oas.yaml example be extracted to a common component for re-use?

Reply:
**question:** The operation schema here doesn't utilize oneOf error/response the same way the protobuf schema does. Is this deliberate or an oversight? Is there a limit to the usefulness of oneOf I'm unaware of?

Reply:
that's an oversight, I'd guess - this PR was primarily meant to just fix the duplicate status code guidance, but we would maybe scope the PR bigger and just to re-review /fix all the issues you mentioned. Would you prefer that?

Reply:
No preference. I'm generally OK with things noticed in PR but out of scope becoming issues to be resolved later or follow on PRs. Or not resolved at all if I'm seeing problems where there are none.